src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1138,4 +1123,45 @@ def users_path
   def add_path(action)
     url_for(action: action, project: @project, role: params[:role], userid: params[:userid], package: @package)
   end
+
+  # Basically backend stores date in /source (package sources) and /build (package
+  # build related). Logically build logs are stored in /build. Though build logs also
+  # contain information related to source packages.
+  # Thus before giving access to the build log, we need to ensure user has source access
+  # rights.
+  #
+  # This before_filter checks source permissions for packages that belong to remote projects,
+  # to local projects and local projects that link to other project's packages.
+  #
+  # If the check succeeds it sets @project and @package variables.
+  def check_build_log_access
+    if Project.exists_by_name(params[:project])
+      @project = Project.get_by_name(params[:project])
+    else
+      redirect_to root_path, error: "Couldn't find project '#{params[:project]}'. Are you sure it still exists?"
+      return false
+    end
+
+    begin
+      @package = Package.get_by_project_and_name(@project, params[:package], { use_source:           false,
+                                                                               follow_multibuild:    true,
+                                                                               follow_project_links: true })
+    rescue Package::UnknownObjectError
+      redirect_to project_show_path(@project.to_param),
+                  error: "Couldn't find package '#{params[:package]}' in " +
+                         "project '#{@project.to_param}'. Are you sure it exists?"
+      return false
+    end
+
+    # package is nil for remote projects
+    if @package && !@package.check_source_access?
+      redirect_to package_show_path(project: @project.name, package: @package.name),
+                  error: "Could not access build log"
```

Comment:
I see... The code was basically there before. So I'm fine with it. Just wanted to mention that.